Assets/Scripts/WorldGenTestScripts/WorldGenerator.cs

```diff
@@ -2,75 +2,182 @@
-            return;
-        for (int x = (int)point1.x; x < (int)point2.x; x++)
+        bool connectionOrientation = chooseConnectionOrientation(connectionsX, connectionsY);
+        Vector2 connector = ChooseConnection(connectionsX, connectionsY, connectionOrientation).GetPosition();
+        Vector2 startPoint = rooms[roomID].HasPlace(pixels, connector, connectionOrientation);
+        if (startPoint != ERROR_CODE)
         {
-            for (int y = (int)point1.y; y < (int)point2.y; y++)
+            Pixel[] newConnections = rooms[roomID].Place(pixels, startPoint);
+            ManageNewConnections(connectionsX, connectionsY, newConnections);
+            return true;
+        }
+        return false;
+    }
+
+    private static void ManageNewConnections(List<Pixel> connectionsX, List<Pixel> connectionsY, Pixel[] newConnections)
+    {
+        for (int k = 0; k < newConnections.Length; k++)
+        {
+            if (newConnections[k].GetState() == State.CORIDOR_X)
+                connectionsX.Add(newConnections[k]);
+            if (newConnections[k].GetState() == State.CORIDOR_Y)
+                connectionsY.Add(newConnections[k]);
+        }
+    }
+
+    private static Pixel ChooseConnection(List<Pixel> connectionsX, List<Pixel> connectionsY, bool horizontalConnecting)
+    {
+        return horizontalConnecting ? connectionsX[random.Next(connectionsX.Count)] :
+            connectionsY[random.Next(connectionsY.Count)];
+    }
+
+    private static bool chooseConnectionOrientation(List<Pixel> connectionsX, List<Pixel> connectionsY)
+    {
+        return connectionsX.Count * connectionsY.Count != 0 ? (random.Next(2) % 2 == 1) : connectionsY.Count != 0 ? false : true;
+    }
+
+    private static bool anyConnections(List<Pixel> connectionsX, List<Pixel> connectionsY)
+    {
+        return connectionsX.Count != 0 || connectionsY.Count != 0;
+    }
+
+    private void GenerateTiles()
+    {
+        for (int i = 0; i < width; i++)
+        {
+            for (int j = 0; j < height; j++)
+            {
+                pixels[i, j].GenerateTile(transform);
+            }
+        }
+    }
+
+    void CreateRoomBorders() 
+    {
+        for (int i = 0; i < width; i++)
+        {
+            for (int j = 0; j < height; j++)
+            {
+                if (IsAreaOccupied(new Vector2(i, j), new Vector2(i, j), 1) &&
+                    pixels[i, j].GetState() == State.EMPTY &&
+                    !OnWorldBorder(new Vector2(i,j)))
+                {
+                    ChangePixelColor(new Vector2(i,j), State.BORDER);
+                }
+            }
+        }
+    }
+
+    void RemoveDeadEnds()
+    {
+        for (int i = 0; i < width; i++)
+        {
+            for (int j = 0; j < height; j++)
             {
-                ChangePixelColor(new Vector2((float)x, (float)y), 2);
+                if ((pixels[i, j].GetState() == State.CORIDOR_X || pixels[i, j].GetState() == State.CORIDOR_Y)
```

Comment:
multiple array accesses and calls to GetState. store to variable
```suggestion
                State state = pixels[i, j].GetState();
                if ((state == State.CORIDOR_X || state == State.CORIDOR_Y)
```